aea/helpers/dialogue/base.py

```diff
@@ -412,21 +362,25 @@ def last_message(self) -> Optional[Message]:
 
         return last_message
 
-    def get_message(self, message_id_to_find: int) -> Optional[Message]:
+    def get_message(self, message_id: int) -> Message:
         """
         Get the message whose id is 'message_id'.
 
-        :param message_id_to_find: the id of the message
+        :param message_id: the id of the message
         :return: the message if it exists, None otherwise
         """
-        result = None  # type: Optional[Message]
+        assert 1 <= message_id, "message_id must be greater than or equal to 1."
+
         list_of_all_messages = self._outgoing_messages + self._incoming_messages
-        for message in list_of_all_messages:
-            if message.message_id == message_id_to_find:
-                result = message
-                break
+        assert message_id <= len(
+            list_of_all_messages
+        ), "This dialogue does not have a message with id {}. Last message id is {}.".format(
+            message_id, len(list_of_all_messages)
+        )
 
-        return result
+        for message in list_of_all_messages:
+            if message.message_id == message_id:
+                return message
```

Comment:
do you really want this to throw if the message_id does not make sense; why? isn't it better to return None?

Reply:
You are supposed to use `get_message` with a correct `message_id`. If it sometimes returns None and sometimes Message, one must always check that it does not return None in every correct use.

I added two other methods `has_message_id` and `has_method` to check whether a message exists in the dialogue.

Reply:
Ok